File: java/android_level_2/app/src/main/java/com/bliepy/MainActivity.java
package com.bliepy;

import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.widget.Toast;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView mRecyclerView;
    private SwipeAdapter mAdapterSwipe;

    private RecyclerView.LayoutManager mLayoutManager;

    private List<SwipeImage> mSwipe;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.content_main);

        mSwipe = SwipeImage.demoData();

        mRecyclerView = findViewById(R.id.recyclerView);

        mLayoutManager = new LinearLayoutManager(this);
        mRecyclerView.setLayoutManager(mLayoutManager);

        mAdapterSwipe = new SwipeAdapter(mSwipe);
        mRecyclerView.setAdapter(mAdapterSwipe);


        ItemTouchHelper.SimpleCallback touch = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT) {

            @Override
            public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder

                    target) {

                return false;

            }

            @Override
            public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction) {
                //Get the index corresponding to the selected position
                int position = (viewHolder.getAdapterPosition());
                SwipeImage image = mSwipe.get(position);

                if (image.isImageEurope()) {
                    if (direction == ItemTouchHelper.LEFT ) {
                        Toast.makeText(MainActivity.this, "GOOD", Toast.LENGTH_SHORT).show();
                    } else {
                        Toast.makeText(MainActivity.this, "WRONG", Toast.LENGTH_SHORT).show();
                    }

                } else {

                    if (direction != ItemTouchHelper.RIGHT ) {
                        Toast.makeText(MainActivity.this, "GOOD", Toast.LENGTH_SHORT).show();
                    } else {
                        Toast.makeText(MainActivity.this, "WRONG", Toast.LENGTH_SHORT).show();
                    }
                }

                mSwipe.remove(position);
                mAdapterSwipe.notifyItemRemoved(position);
            }
        };

        ItemTouchHelper itemTouchHelper = new ItemTouchHelper(touch);
        itemTouchHelper.attachToRecyclerView(mRecyclerView);
        updateUI();
    }

    private void updateUI() {

        if (mAdapterSwipe == null) {
            mAdapterSwipe = new SwipeAdapter(mSwipe);
            mRecyclerView.setAdapter(mAdapterSwipe);

        } else {
            mAdapterSwipe.notifyDataSetChanged();
        }
    }
}

File: kotlin/level_2_task_2/app/src/main/java/com/bliepy/swipeapp/MainActivity.kt
package com.bliepy.swipeapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.AdapterView
import android.widget.Toast
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {


    private val questionList = listOf(
        Question("A 'val' and 'var' are the same.", false),
        Question("Mobile Application Development grants 12 ETCS.", true),
        Question("A Unit in Kotlin corresponds to a void in Java", true),
        Question("In Kotlin 'when' replaces the 'switch' operator in Java", true)
    )

    private val questionAdapter = QuestionAdapter(questionList) { item : Question -> questionItemClicked(item) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rvReminders : RecyclerView = findViewById(R.id.questionList)

        rvReminders.layoutManager = LinearLayoutManager(this@MainActivity, RecyclerView.VERTICAL, false)
        rvReminders.adapter = questionAdapter
        rvReminders.addItemDecoration(DividerItemDecoration(this@MainActivity, DividerItemDecoration.VERTICAL))

        val callbackSwipeRight = object: ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT) {
            override fun onMove(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder,
                target: RecyclerView.ViewHolder
            ): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
                val message = questionList[viewHolder.adapterPosition].answer
                if (message) {
                    Toast.makeText(this@MainActivity, "TRUE", Toast.LENGTH_SHORT).show()
                }else{
                    Toast.makeText(this@MainActivity, "FALSE", Toast.LENGTH_SHORT).show()
                }
                questionAdapter.notifyItemChanged(viewHolder.adapterPosition)
            }
        }


        val callbackSwipeLeft = object: ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.LEFT) {

            override fun onMove(
                recyclerView: RecyclerView,
                viewHolder: RecyclerView.ViewHolder,
                target: RecyclerView.ViewHolder
            ): Boolean {
                return false
            }

            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
                val message = questionList[viewHolder.adapterPosition].answer
                if (message) {
                    Toast.makeText(this@MainActivity, "TRUE", Toast.LENGTH_SHORT).show()
                }else{
                    Toast.makeText(this@MainActivity, "FALSE", Toast.LENGTH_SHORT).show()
                }
                questionAdapter.notifyItemChanged(viewHolder.adapterPosition)
            }

        }

        ItemTouchHelper(callbackSwipeLeft).attachToRecyclerView(rvReminders)
        ItemTouchHelper(callbackSwipeRight).attachToRecyclerView(rvReminders)
    }

    private fun questionItemClicked(item : Question) {
        Toast.makeText(this, "The answer: ${item.answer}", Toast.LENGTH_SHORT).show()
    }
}
